src/flint/fmpq_mpoly.jl

```diff
@@ -760,6 +765,20 @@ sub!(a::QQMPolyRingElem, b::RationalUnion, c::QQMPolyRingElem) = neg!(sub!(a, c,
 mul!(a::QQMPolyRingElem, b::QQMPolyRingElem, c::RationalUnion) = mul!(a, b, flintify(c))
 mul!(a::QQMPolyRingElem, b::RationalUnion, c::QQMPolyRingElem) = mul!(a, c, b)
 
+# special: multiply a QQMPolyRingElem by an integer and store the result as a ZZMPolyRingElem.
+# obviously this only works if the integers is a multiple of the denominator of the polynomial
+function mul!(a::ZZMPolyRingElem, b::QQMPolyRingElem, c::IntegerUnion)
+  x = QQFieldElem()
+  GC.@preserve a b x begin
+    x = mul!(x, _content_ptr(b), c)
+    bp = _zpoly_ptr(b)
```

Comment:
```suggestion
	@assert is_integer(x)
    bp = _zpoly_ptr(b)
```
I know that this is an unsafe function, but I think adding something like this would still avoid weird usage issues in the long run.
The problem being that this does not error or corrupt memory (which usually errors at some future point), but instead silently computes a wrong result.

Reply:
I don't understand the distinction. How is garbage output due to corrupt memory (= seemingly random weird stuff, from wrong results to crashes to "everything seems to work") "better" than ignoring a denominator 'as requested' in a fully deterministic way?

But that said, I really don't mind having such an `@assert` in (I had one initially).

But I was under the impression that @thofma think there should be either no such `@assert` or else multiple for everything (but I didn't see a point in checking e.g. internal orderings for a match when no other `mul!` or `add!` etc. method does). Or maybe I misunderstood that?

Reply:
yeah, true. If people use this, they should be sure what they are doing.